src/linpde_gp/linfuncops/diffops/_coefficients.py

```diff
@@ -0,0 +1,144 @@
+from copy import deepcopy
+from typing import Dict, Tuple
+
+import numpy as np
+from probnum.typing import ShapeType
+
+
+class PartialDerivativeCoefficients:
+    r"""Partial derivative coefficients of a linear differential operator.
+
+    Any linear differential operator can be written as a sum of partial derivatives.
+
+    The coefficients are stored in a dictionary of the form
+    {input_codomain_index: {multi_index: coefficient}}.
+
+    For example, the Laplacian operator in 2D can be written as
+
+    .. math::
+        \Delta = \frac{\partial^2}{\partial x^2} + \frac{\partial^2}{\partial y^2}
+
+    and is stored as
+
+    .. code-block:: python
+
+        {
+            (): {
+                (2, 0): 1.0,
+                (0, 2): 1.0,
+            }
+        }
+
+    Parameters
+    ----------
+    coefficient_dict :
+        Dictionary of coefficients.
+    """
+
+    def __init__(
+        self, coefficient_dict: Dict[ShapeType, Dict[Tuple[int, ...], float]]
+    ) -> None:
+        self._input_codomain_shape_bound = None
```

Comment:
I think we should pass `domain_shape` and `input_codomain_shape` as constructor parameters to ensure the integrity of the datastructure. That way we can also make `validate_input_{co}domain_shape` part of the constructor.